### PR TITLE
mission system component

### PR DESCRIPTION
**Pull request by TheEmidee**
- Replaced mission system by a component
- Removed unused functions
- Give mission actions an outer for GetWorld()

### CODE REVIEW

File: Source/MissionSystem/Private/MSMissionSystemComponent.cpp [hunk 1]
```diff
@@ -1,542 +1,575 @@
...
-        {
-            observer.Callback.ExecuteIfBound( objective->GetClass(), was_cancelled );
-            MissionObjectiveEndObservers.RemoveAt( index );
-        }
-    }
-
-    ViewModel->SetMissionObjectiveEnded( mission, objective );
+#include "MSMissionSystemComponent.h"
+
+#include "Engine/GameInstance.h"
```
Comment (Frickerson) [inline Source/MissionSystem/Private/MSMissionSystemComponent.cpp]: <>

File: Source/MissionSystem/Private/MSMissionSystemComponent.cpp [hunk 2]
```diff
@@ -1,542 +1,575 @@
...
+
+#include <Engine/World.h>
+#include <Serialization/MemoryWriter.h>
+
+#if !( UE_BUILD_SHIPPING || UE_BUILD_TEST )
+static FAutoConsoleCommand SkipMissionsCommand(
+    TEXT( "MissionSystem.SkipMissions" ),
+    TEXT( "Skips the current missions." ),
+    FConsoleCommandWithWorldArgsAndOutputDeviceDelegate::CreateLambda( []( const TArray< FString > & /*args*/, const UWorld * world, FOutputDevice & /*output_device*/ ) {
+        /*if ( const auto * mission_system = world->GetSubsystem< UMSMissionSystemComponent >() )
```
Comment (Frickerson) [inline Source/MissionSystem/Private/MSMissionSystemComponent.cpp:19]: Should maybe add a todo for all these commented functions?

**Review by Frickerson — CHANGES_REQUESTED** [review]

File: Source/MissionSystem/Private/MSMissionSystemComponent.cpp (hunk 1 above)
Comment (TheEmidee) [reply Source/MissionSystem/Private/MSMissionSystemComponent.cpp]: done

File: Source/MissionSystem/Private/MSMissionSystemComponent.cpp (hunk 2 above)
Comment (TheEmidee) [reply Source/MissionSystem/Private/MSMissionSystemComponent.cpp:19]: done

**Pull request merged**